vite.config.js

```diff
@@ -0,0 +1,48 @@
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+import path from 'path'
+import { peerDependencies, dependencies } from './package.json';
+
+const externalPackages = [
+  ...Object.keys(dependencies || {}),
+  ...Object.keys(peerDependencies || {}),
+];
+
+const regexesOfPackages = externalPackages.map(
+  (packageName) => new RegExp(`^${packageName}(/.*)?`),
+);
+
+export default defineConfig({
+  plugins: [react({
+    babel: {
+      configFile: true,
+    },
+  })],
+  build: {
+    lib: {
+      entry: path.resolve(__dirname, './src/index.js'),
+      formats: ['es', 'cjs']
```

Comment:
This warning suggest we can drop this line:
```
"build.lib.formats" will be ignored because "build.rollupOptions.output" is already an array format
```

```suggestion
```

Reply:
But then it will complain, that I have to provide an name to the bundle as it required by IIFE which is default options in `formats`
I will try to add name, nope it won't break anything

Reply:
@tzimmermann seems to be ok with name and default options